migrate/v0_13/migrate.go

```diff
@@ -78,6 +89,211 @@ func MigrateCDP(oldGenState v0_11cdp.GenesisState) v0_13cdp.GenesisState {
+	for _, rp := range hardGS.Params.LiquidityProviderSchedules {
+		newRP := v0_13incentive.NewMultiRewardPeriod(true, rp.DepositDenom, rp.Start, rp.End, sdk.NewCoins(rp.RewardsPerSecond))
+		hardSupplyRewardPeriods = append(hardSupplyRewardPeriods, newRP)
+	}
+	hardBorrowRewardPeriods := v0_13incentive.MultiRewardPeriods{}
+	hardDelegatorRewardPeriods := v0_13incentive.RewardPeriods{}
+	for _, rp := range hardGS.Params.DelegatorDistributionSchedules {
+		newRP := v0_13incentive.NewRewardPeriod(rp.DistributionSchedule.Active, rp.DistributionSchedule.DepositDenom, rp.DistributionSchedule.Start, rp.DistributionSchedule.End, rp.DistributionSchedule.RewardsPerSecond)
+		hardDelegatorRewardPeriods = append(hardDelegatorRewardPeriods, newRP)
+	}
+	params := v0_13incentive.NewParams(usdxMintingRewardPeriods, hardSupplyRewardPeriods, hardBorrowRewardPeriods, hardDelegatorRewardPeriods, v0_13incentive.Multipliers{v0_13incentive.NewMultiplier(v0_13incentive.Small, 1, sdk.MustNewDecFromStr("0.25")), v0_13incentive.NewMultiplier(v0_13incentive.Large, 12, sdk.MustNewDecFromStr("1.0"))}, ClaimEndTime)
+
+	usdxGenAccumulationTimes := v0_13incentive.GenesisAccumulationTimes{}
+
+	for _, rp := range params.USDXMintingRewardPeriods {
+		gat := v0_13incentive.NewGenesisAccumulationTime(rp.CollateralType, GenesisTime, sdk.ZeroDec())
+		usdxGenAccumulationTimes = append(usdxGenAccumulationTimes, gat)
+	}
+	hardSupplyGenAccumulationTimes := v0_13incentive.GenesisAccumulationTimes{}
+	for _, rp := range params.HardSupplyRewardPeriods {
+		gat := v0_13incentive.NewGenesisAccumulationTime(rp.CollateralType, GenesisTime, sdk.ZeroDec())
+		hardSupplyGenAccumulationTimes = append(hardSupplyGenAccumulationTimes, gat)
+	}
+	hardBorrowGenAccumulationTimes := v0_13incentive.GenesisAccumulationTimes{}
+	for _, rp := range params.HardBorrowRewardPeriods {
+		gat := v0_13incentive.NewGenesisAccumulationTime(rp.CollateralType, GenesisTime, sdk.ZeroDec())
+		hardBorrowGenAccumulationTimes = append(hardBorrowGenAccumulationTimes, gat)
+	}
+
+	hardDelegatorGenAccumulationTimes := v0_13incentive.GenesisAccumulationTimes{}
+
+	for _, rp := range params.HardDelegatorRewardPeriods {
+		gat := v0_13incentive.NewGenesisAccumulationTime(rp.CollateralType, GenesisTime, sdk.ZeroDec())
+		hardDelegatorGenAccumulationTimes = append(hardDelegatorGenAccumulationTimes, gat)
+	}
+
+	usdxClaims := v0_13incentive.USDXMintingClaims{}
+	usdxClaimMap := make(map[string]v0_13incentive.USDXMintingClaim)
+	claimEndMap := make(map[uint64]time.Time)
+
+	for _, cp := range incentiveGS.ClaimPeriods {
+		claimEndMap[cp.ID] = cp.End
+	}
+
+	for _, claim := range incentiveGS.Claims {
+		if claimEndMap[claim.ClaimPeriodID].Before(GenesisTime) {
+			continue
+		}
+		newClaim, ok := usdxClaimMap[claim.Owner.String()]
+		if !ok {
+			newClaim = v0_13incentive.NewUSDXMintingClaim(claim.Owner, claim.Reward, v0_13incentive.RewardIndexes{v0_13incentive.NewRewardIndex(claim.CollateralType, sdk.ZeroDec())})
```

Comment:
note: the migration will need to be updated again when we move `USDXMintingClaim.RewardIndexes` and `HardLiquidityProviderClaim.DelegatorIndexes` to type `MultiRewardIndex`

Reply:
This suggestion is no longer applicable as we won't be updating `USDXMintingClaim.RewardIndexes` and `HardLiquidityProviderClaim.DelegatorIndexes` to `MultiRewardIndex` for kava-5.